driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -21,67 +21,76 @@
 import com.mongodb.MongoException;
 import com.mongodb.ServerAddress;
 import com.mongodb.ServerCursor;
+import com.mongodb.binding.ConnectionSource;
 import com.mongodb.binding.ReadBinding;
 import org.bson.BsonDocument;
+import org.bson.BsonTimestamp;
 import org.bson.RawBsonDocument;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import static com.mongodb.operation.ChangeStreamBatchCursorHelper.isRetryableError;
 
-final class ChangeStreamBatchCursor<T> implements BatchCursor<T> {
+final class ChangeStreamBatchCursor<T> implements AggregateResponseBatchCursor<T> {
     private final ReadBinding binding;
     private final ChangeStreamOperation<T> changeStreamOperation;
 
+    private AggregateResponseBatchCursor<RawBsonDocument> wrapped;
     private BsonDocument resumeToken;
-    private BatchCursor<RawBsonDocument> wrapped;
 
     ChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation,
-                            final BatchCursor<RawBsonDocument> wrapped,
+                            final AggregateResponseBatchCursor<RawBsonDocument> wrapped,
                             final ReadBinding binding) {
         this.changeStreamOperation = changeStreamOperation;
-        this.resumeToken = changeStreamOperation.getResumeToken();
+        this.wrapped = wrapped;
+        this.binding = binding.retain();
         if (changeStreamOperation.getStartAfter() != null) {
-            this.resumeToken = changeStreamOperation.getStartAfter();
-        }
-        if (this.resumeToken == null) {
+            resumeToken = changeStreamOperation.getStartAfter();
+        } else if (changeStreamOperation.getResumeAfter() != null) {
+            resumeToken = changeStreamOperation.getResumeAfter();
+        } else if (changeStreamOperation.getStartAtOperationTime() == null && getMaxWireVersion() >= 7
```

Comment:
Could you please explain the last two `if-else` a little bit?

Reply:
In this `else-if` block, we save the operation time from the aggregate response if `startAtOperationTime` has not been set, the max wire version is >= 7, and no postBatchResumeToken was provided in the aggregate response. 
In the last `else-if` block, if no resume token has been found in the change stream options, we set `startOperationTimeForResume` with the operation time from the binding.

Reply:
I have removed the setting of `startOperationTimeForResume` in the new ChangeStreamOperation option setting methods.